File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/RowBasedCommand_ViewModel.cpp
#include "RowBasedCommand_ViewModel.h"
#include <vector>
#include <memory>
#include "RowBasedCommand_ViewModelMyFlagsRow.h"
#include "RowBasedCommand_ViewModelMyComboBoxesRow.h"
#include "RowBasedCommand_ViewModelMyStandardControlsRow.h"
#include "RowBasedCommand_ViewModelRowBasedCommandWithNoRowHandleParameterRow.h"
#include <string>

namespace commands
{
  std::vector<std::shared_ptr<RowBasedCommand_ViewModelMyFlagsRow>> RowBasedCommand_ViewModel::getMyFlagsListRows()
  {
    return this->MyFlagsListRows;
  }
  std::vector<std::shared_ptr<RowBasedCommand_ViewModelMyComboBoxesRow>> RowBasedCommand_ViewModel::getMyComboBoxesTableRows()
  {
    return this->MyComboBoxesTableRows;
  }
  std::vector<std::shared_ptr<RowBasedCommand_ViewModelMyStandardControlsRow>> RowBasedCommand_ViewModel::getMyStandardControlsTreeRows()
  {
    return this->MyStandardControlsTreeRows;
  }
  std::vector<std::shared_ptr<RowBasedCommand_ViewModelRowBasedCommandWithNoRowHandleParameterRow>> RowBasedCommand_ViewModel::getRowBasedCommandWithNoRowHandleParameterListRows()
  {
    return this->RowBasedCommandWithNoRowHandleParameterListRows;
  }
  int RowBasedCommand_ViewModel::getRowBasedCommandWithNoRowHandleParameterListSelectedRow()
  {
    return this->RowBasedCommandWithNoRowHandleParameterListSelectedRow;
  }
  void RowBasedCommand_ViewModel::loadViewModel()
  {
  }
  void RowBasedCommand_ViewModel::checkboxesCheckBoxChecked(int rowIndex, bool isChecked)
  {
  }
  void RowBasedCommand_ViewModel::freeTextComboBoxesComboBoxEntrySelected(int rowIndex, std::string entryName)
  {
  }
  void RowBasedCommand_ViewModel::entryComboBoxesComboBoxEntrySelected(int rowIndex, std::string entryName)
  {
  }
  void RowBasedCommand_ViewModel::textBoxesTextBoxTextFilled(int rowIndex, std::string text)
  {
  }
  void RowBasedCommand_ViewModel::buttonsButtonClicked(int rowIndex)
  {
  }
  void RowBasedCommand_ViewModel::checkboxesCheckBoxChecked(bool isChecked)
  {
  }
  void RowBasedCommand_ViewModel::rowBasedCommandWithNoRowHandleParameterListRowSelected(int rowIndex)
  {
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/RowBasedCommand_ViewModel.cs
/// <filename>
///     RowBasedCommand_ViewModel.cs
/// </filename>
using System.Collections.Generic;
using commands;

namespace commands
{
    public class RowBasedCommand_ViewModel
    {
        private System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyFlagsRow> MyFlagsListRows = new System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyFlagsRow>();

        public virtual System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyFlagsRow> getMyFlagsListRows()
        {
            return this.MyFlagsListRows;
        }

        private System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyComboBoxesRow> MyComboBoxesTableRows = new System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyComboBoxesRow>();

        public virtual System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyComboBoxesRow> getMyComboBoxesTableRows()
        {
            return this.MyComboBoxesTableRows;
        }

        private System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyStandardControlsRow> MyStandardControlsTreeRows = new System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyStandardControlsRow>();

        public virtual System.Collections.Generic.List<commands.RowBasedCommand_ViewModelMyStandardControlsRow> getMyStandardControlsTreeRows()
        {
            return this.MyStandardControlsTreeRows;
        }

        private System.Collections.Generic.List<commands.RowBasedCommand_ViewModelRowBasedCommandWithNoRowHandleParameterRow> RowBasedCommandWithNoRowHandleParameterListRows = new System.Collections.Generic.List<commands.RowBasedCommand_ViewModelRowBasedCommandWithNoRowHandleParameterRow>();

        public virtual System.Collections.Generic.List<commands.RowBasedCommand_ViewModelRowBasedCommandWithNoRowHandleParameterRow> getRowBasedCommandWithNoRowHandleParameterListRows()
        {
            return this.RowBasedCommandWithNoRowHandleParameterListRows;
        }

        private int RowBasedCommandWithNoRowHandleParameterListSelectedRow;

        public virtual int getRowBasedCommandWithNoRowHandleParameterListSelectedRow()
        {
            return this.RowBasedCommandWithNoRowHandleParameterListSelectedRow;
        }

        public virtual void loadViewModel()
        {
        }

        public virtual void checkboxesCheckBoxChecked(int rowIndex, bool isChecked)
        {
        }

        public virtual void freeTextComboBoxesComboBoxEntrySelected(int rowIndex, string entryName)
        {
        }

        public virtual void entryComboBoxesComboBoxEntrySelected(int rowIndex, string entryName)
        {
        }

        public virtual void textBoxesTextBoxTextFilled(int rowIndex, string text)
        {
        }

        public virtual void buttonsButtonClicked(int rowIndex)
        {
        }

        public virtual void checkboxesCheckBoxChecked(bool isChecked)
        {
        }

        public virtual void rowBasedCommandWithNoRowHandleParameterListRowSelected(int rowIndex)
        {
        }
    }
}
